entity-service/src/integrationTest/java/org/hypertrace/entity/service/service/EntityDataServiceTest.java

```diff
@@ -874,6 +875,167 @@ public void testEntityQueryOrderBy() {
+        .setName("attributes.labels")
+        .setOperator(Operator.NOT_IN)
+        .setAttributeValue(AttributeValue.newBuilder()
+            .setValueList(
+                AttributeValueList.newBuilder()
+                    .addValues(
+                        AttributeValue.newBuilder().setValue(Value.newBuilder().setString("b1"))
+                    )
+                    .addValues(
+                        AttributeValue.newBuilder().setValue(Value.newBuilder().setString("v4"))
+                    )
+            )
+        ).build();
+
+    Query query = Query.newBuilder()
+        .setEntityType(EntityType.K8S_POD.name())
+        .setFilter(attributeFilter).build();
+    List<Entity> entitiesList = entityDataServiceClient.query(TENANT_ID, query);
+    assertTrue(entitiesList.size() == 3);
+    assertTrue(entitiesList.contains(createdEntity1)
+        && entitiesList.contains(createdEntity2)
+        && entitiesList.contains(createdEntity5));
+
+    // test IN
+    attributeFilter = AttributeFilter.newBuilder()
+        .setName("attributes.labels")
+        .setOperator(Operator.IN)
+        .setAttributeValue(AttributeValue.newBuilder()
+            .setValueList(
+                AttributeValueList.newBuilder()
+                    .addValues(
+                        AttributeValue.newBuilder().setValue(Value.newBuilder().setString("b1"))
+                    )
+                    .addValues(
+                        AttributeValue.newBuilder().setValue(Value.newBuilder().setString("v4"))
+                    )
+            )
+        ).build();
+
+    query = Query.newBuilder().setEntityType(EntityType.K8S_POD.name()).setFilter(attributeFilter)
+        .build();
+    entitiesList = entityDataServiceClient.query(TENANT_ID, query);
+    assertTrue(entitiesList.size() == 2);
+    assertTrue(entitiesList.contains(createdEntity3) && entitiesList.contains(createdEntity4));
+
+    // test NEQ
+    attributeFilter = AttributeFilter.newBuilder().setOperator(Operator.AND)
+        .addChildFilter(
+            AttributeFilter.newBuilder()
+                .setName("attributes.labels")
+                .setOperator(Operator.NEQ)
+                .setAttributeValue(AttributeValue.newBuilder()
+                    .setValue(Value.newBuilder().setString("v2"))
+                )
+        )
+        .addChildFilter(
+            AttributeFilter.newBuilder()
+                .setName("attributes.labels")
+                .setOperator(Operator.NEQ)
+                .setAttributeValue(AttributeValue.newBuilder()
+                    .setValue(Value.newBuilder().setString("b1"))
+                )
+        ).build();
+
+    query = Query.newBuilder().setEntityType(EntityType.K8S_POD.name()).setFilter(attributeFilter)
+        .build();
+    entitiesList = entityDataServiceClient.query(TENANT_ID, query);
+    assertTrue(entitiesList.size() == 2);
+    assertTrue(entitiesList.contains(createdEntity2) && entitiesList.contains(createdEntity2));
```

Comment:
I think there is a typo here 

```
entitiesList.contains(createdEntity2) && entitiesList.contains(createdEntity2)
```

Reply:
I think it should be 

```
entitiesList.contains(createdEntity2) && entitiesList.contains(createdEntity5)
```

Reply:
Done. Good catch, while debugging, I had checked `createdEntity5`, but had gone back & forth changing this condition.